### PR TITLE
Implement debounce for guitar

### LINKED ISSUE #17: Debounce guitar strumming
The strumming paddle now sometimes sends multiple NoteOn/Off evens in quick succession. We should probably debounce for a couple of millis to avoid this.

### PR DESCRIPTION
**Pull request by laurynaslubys**
Fixes #17

### CODE REVIEW

File: guitar-midi/src/main.cpp [hunk 1]
```diff
@@ -100,13 +108,17 @@ void setup()
...
+  updateButton(PIN_GREEN,  48, now, pinState[0], debounceState[0]);
+  updateButton(PIN_RED,    49, now, pinState[1], debounceState[1]);
+  updateButton(PIN_YELLOW, 50, now, pinState[2], debounceState[2]);
+  updateButton(PIN_BLUE,   51, now, pinState[3], debounceState[3]);
+  updateButton(PIN_ORANGE, 52, now, pinState[4], debounceState[4]);
+  // Pins UP and down share the same debounce state intentionally.
+  // When strumming with a particular pattern, the button flies back and opens 
+  // the other switch, which is not something we want.
+  updateButton(PIN_UP,     59, now, pinState[5], debounceState[5]); 
+  updateButton(PIN_DOWN,   58, now, pinState[6], debounceState[5]);
```
Comment (bagdonas) [inline guitar-midi/src/main.cpp:121]: Another way to do the same without passing everything around (not sure if it's better or worse... probably much worse, but the copy part is veeeery explicit) 😄

```c
  now = millis(); // global
  updateButton(PIN_GREEN, 48);
  // ...
  updateButton(PIN_ORANGE, 52);
  // Pins UP and DOWN share the same debounce state intentionally.
  // When strumming with a particular pattern, the button flies back and opens 
  // the other switch, which is not something we want.
  updateButton(PIN_UP, 59); 
  debounceState[PIN_DOWN] = debounceState[PIN_UP]; 
  updateButton(PIN_DOWN, 58);
  debounceState[PIN_UP] = debounceState[PIN_DOWN]; 
```

**Review by bagdonas — APPROVED** [review]

**Comment by bagdonas** [comment]
Not sure what exact value of `A3` is, it could be in range 0..6, in which case it could overwrite button state... I guess `updateSelector` should be reworked to also receive pointer to `pinState` element at some explicitly defined index.

File: guitar-midi/src/main.cpp (hunk 1 above)
Comment (laurynaslubys) [reply guitar-midi/src/main.cpp]: I think it's nicer to explicitly pass in state. That said, I modified this to be more explicit, see if it looks better.

Comment (bagdonas) [reply guitar-midi/src/main.cpp]: nice

**Pull request merged**